deploy/charts/csi-driver/templates/clusterrole.yaml

```diff
@@ -8,3 +8,12 @@ rules:
 - apiGroups: ["cert-manager.io"]
   resources: ["certificaterequests"]
   verbs: ["watch", "create", "delete", "list"]
+{{- $securityContextConstraintImplicitlyEnabled := and (kindIs "string" .Values.openshift.securityContextConstraint.enabled) (eq .Values.openshift.securityContextConstraint.enabled "detect") (.Capabilities.APIVersions.Has "security.openshift.io/v1") }}
+{{- $securityContextConstraintExplicitlyEnabled := and (kindIs "bool" .Values.openshift.securityContextConstraint.enabled) (.Values.openshift.securityContextConstraint.enabled) }}
+{{- $securityContextConstraintEnabled := or $securityContextConstraintImplicitlyEnabled $securityContextConstraintExplicitlyEnabled }}
```

Comment:
nitpick: I think this logic makes sense to me but given it's Helm template logic it's quite _busy_.

Could we add some template comments here in this file explaining what the logic is doing and why? I imagine users will quite often check the roles / clusterroles in a chart to see what permissions are enabled and especially for permissions which have "security" in them it would be helpful to be really clear about what's going on in the template